step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/bucket/BucketBuilder.java

```diff
@@ -58,6 +60,12 @@ public BucketBuilder accumulate(Bucket bucket) {
             bucketDistribution.forEach((key, value) ->
                     distribution.computeIfAbsent(key, k -> new LongAdder()).add(value));
         }
+        BucketAttributes bucketAttr = bucket.getAttributes();
+        if (bucketAttr != null) {
+            bucketAttr.forEach((k, v) -> {
+                ((HashSet<Object>) attributes.computeIfAbsent(k, i -> new HashSet<>())).add(v);
```

Comment:
@david-stephan For large time ranges the number of attributes could be quite high. Shouldn't we add an option to enable/disable this feature?
Shouldn't we also limit the number of  values in the HashSet to avoid loading large amount of data into memory?
We're kind of misusing the BucketAttributes map to return a Set for each entry value where a single value would be expected. In my opinion this OK and makes also in a certain way but should be clearly documented in the aggregation pipeline.